test/testcondition.cpp

```diff
@@ -655,6 +655,13 @@ class TestCondition : public TestFixture {
               "}");
         ASSERT_EQUALS("", errout.str());
 
+        //TRAC #7567: "(a | 7) > 6U" is always true for unsigned a
+        check("void f(unsigned int a) {\n"
+              "assert( (a | 0x7) > 6U );\n" // statement always true for unsigned a
```

Comment:
and also add testcases when '&' is used. I mostly want to see that there are not false positives.